### PR TITLE
CONTRIBUTING: a first pass at this document

### PR DESCRIPTION
**Pull request by rjbs**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+This document is meant to be a quick overview of the most important things you
+need to know to get your work reviewed, approved, and into Cyrus.
+
+## Your Code
+
+Cyrus doesn't have a hard and fast style guide, *but it will*.  For now,
+consult the [Cyrus hacking
+docs](https://www.cyrusimap.org/imap/developer/guidance/hacking.html), which
+spell out some of the standards of formatting and construction.
```
Comment (elliefm) [inline CONTRIBUTING.md:19]: I wonder if we should say something about "copy the style of the surrounding code"?  I feel like this should go without saying, and yet.

I feel like this PR should also add a note: or warning: box to the top of hacking.rst explaining that it's very out of date, which might be an alternate place to say "copy the style of the surrounding code".

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  integration test suite.  It can and should be run against your build of
+  Cyrus.  You need to get it from a separate repository.  (Follow that link!)
+* The [cunit tests](https://www.cyrusimap.org/imap/developer/unit-tests.html)
+  are located in the Cyrus IMAP repository, in `./cunit` and run by `make
+  check`.  You should run these, too.
+
+## Submitting Your Work
+
+We use GitHub, including pull requests.  Submit a pull request.  One of the
+committers should review it soon.  If they don't, the best place to ask for
```
Comment (dilyanpalauzov) [inline CONTRIBUTING.md:40]: „Soon“ does not reflect what happens in practice.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+We stop releasing minor releases for major releases after two years.  While we
+might push bugfixes for significant problems to the git branch for an old major
+release, we won't undertake a new release.  If you're running an old version of
+Cyrus, it's up to you (or your package manager) to track and package new
+patches.
+
+If we discover a security vulnerability in a non-development-snapshot version
+of Cyrus, we practice responsible disclosure.  We produce a fix, then inform
+downstream package mangers of that fix, with an embargo date so that the fix
+can be releaed publicly at the same time that updated packages become
```
Comment (dilyanpalauzov) [inline CONTRIBUTING.md:82]: typo: released

Comment (rjbs) [reply CONTRIBUTING.md]: Thanks, amended.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+**All code is reviewed before merge.**  This includes code submitted by
+committers.  This means that if you want to know what awaits you in code
+review, you can look at some recently merged or closed pull requests.
+
+## Cyrus Versioning and Bugfix Policy
+
+Cyrus is free software that comes with no guarantees, but we try to fix bugs
+when they're found.  The policy on that is something like this:
+
+* We release a new **development snapshot** of Cyrus about once a month.  While
```
[the remark this answers is not in the document]
Comment (rjbs) [reply CONTRIBUTING.md:58]: Yeah, I think we should revisit the "make it easy to JFDI this" plan.

File: CONTRIBUTING.md (hunk 1 above)
Comment (rjbs) [reply CONTRIBUTING.md]: I have added that to this PR in both the guide and in this document.

File: CONTRIBUTING.md (hunk 2 above)
Comment (rjbs) [reply CONTRIBUTING.md:43]: Agreed, but the same goes for other things in here.  First we right down what we're shooting for, then we'll keep working on it.

Comment (dilyanpalauzov) [reply CONTRIBUTING.md:43]: From when on will review happen soon?

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+You should run the tests.  Submitting a change that breaks existing tests isn't
+good for anybody!  Adding new tests to assert that your changes work is even
+better, and should be the goal.
+
+There are two kinds of tests:
+
+* The [Cassandane test
+  suite](https://www.cyrusimap.org/imap/developer/developer-testing.html) is an
+  integration test suite.  It can and should be run against your build of
+  Cyrus.  You need to get it from a separate repository.  (Follow that link!)
```
Comment (elliefm) [inline CONTRIBUTING.md:35]: "You need to get it from a separate repository" is no longer true.  The link referenced has already been updated.

**Review by elliefm — APPROVED** [review]
minor nit about the cassandane details, which are no longer current since we integrated the two branches

**Pull request merged**